docs/how-to/rccl-usage-tips.rst

```diff
@@ -82,22 +82,174 @@ set the HSA environment variable as follows:
+NPS2, and NPS4. A compute partition mode can only be combined with specific
+memory partition modes.
+
+- Compute partition modes 
+
+  - SPX - Single partition mode.
+  
+  - DPX - Dual partition mode or two equal spatial partitions.
+
+  - TPX - Triple partition mode or three equal spatial partitions.
+
+  - QPX - Quad partition mode or four equal spatial partitions.
+
+  - CPX - Core partition mode. In this mode, each compute partition
+    mode only has a single core cluster.
+  
+- Memory partition modes 
+
+  - NPS1 - One memory partition/NUMA domain per device.
+
+  - NPS2 - Two memory partitions/NUMA domains per device.
+
+  - NPS4 - Four memory partitions/NUMA domains per device.
+
+For more information, see :doc:`Dynamically change GPU partition modes <rocm:how-to/dynamic_partition_modes>`
+page. The MI300 can be set to CPX mode using the ``rocm-smi`` commands below.
+
+.. code-block:: shell
+
+   sudo rmmod amdgpu
+   sudo rmmod ast
+   sudo modprobe amdgpu
+   rocm-smi --setcomputepartition CPX
+ 
+   ## rocm-smi should list 64 GPUs (devices 0 to 63)
+   rocm-smi
+
+This setting can be also configured with sysfs, as shown in the following example.
+
+.. code-block:: shell
+
+  $ cat /sys/bus/pci/devices/../current_memory_partition
+  NPS1
+  $ cat /sys/bus/pci/devices/../current_compute_partition
+  SPX
+  $ echo NPS4 > /sys/bus/pci/devices/../current_memory_partition
+  $ rmmod amdgpu
+  $ modprobe amdgpu
+  $ cat /sys/bus/pci/devices/../current_memory_partition
+  NPS4
+  $ cat /sys/bus/pci/devices/../current_compute_partition
+  CPX
+
+RCCL performance with CPX and NPS4
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To run RCCL allreduce on 64 GPUs with CPX+NPS4 mode on the MI300X, use this example:
+
+.. code-block:: shell
+
+   mpirun -np 64 --bind-to numa rccl-tests/build/all_reduce_perf -b 8 -e 1G -f 2 -g 1
+
+To run RCCL allreduce on 8 GPUs in the same OAM with CPX+NPS4 mode on the MI300X, use this example:
+
+.. code-block:: shell
+
+   export ROCR_VISIBLE_DEVICES=0,1,2,3,4,5,6,7
+
+   mpirun -np 8 --bind-to numa rccl-tests/build/all_reduce_perf -b 8 -e 1G -f 2 -g 1
+
+RCCL delivers improved allreduce performance in CPX mode for TP=8 (8 GPUs in
+the same OAM) on the MI300X.
+
+.. code-block:: shell
+
+   export HIP_FORCE_DEV_KERNARG=1
+   export RCCL_MSCCLPP_THRESHOLD=1073741824
+
+   export MSCCLPP_READ_ALLRED=1 
+   export ROCR_VISIBLE_DEVICES=0,1,2,3,4,5,6,7
+
+   mpirun -np 8 --bind-to numa rccl-tests/build/all_reduce_perf -b 32 -e 1G -f 2 -g 1 -G 2 -w 20 -n 50
+
+Here are the benchmark results for in-place (where the output buffer is used as the input
+buffer) and out-of-place reduce bus bandwidth.
+
+.. figure:: ../data/how-to/rccl-usage-tips/in-place_reduce.png
+    :alt: In-place reduce benchmark results
+    :align: center
+
+.. figure:: ../data/how-to/rccl-usage-tips/out-of-place_reduce.png
+    :alt: Out-of-place reduce benchmark results
+    :align: center
```

Comment:
Please add a discussion on the performance improvement achieved here.

Reply:
DONE.